pkg/aws/core.go

```diff
@@ -243,6 +247,18 @@ func (d *Driver) CreateMachine(ctx context.Context, req *driver.CreateMachineReq
 		}
 	}
 
+	for i, netIf := range providerSpec.NetworkInterfaces {
+		networkInterfaceID := runResult.Instances[0].NetworkInterfaces[i].NetworkInterfaceId
+		input := &ec2.AssignIpv6AddressesInput{
+			NetworkInterfaceId: networkInterfaceID,
```

Comment:
Do you want to inline `networkInterfaceID`?

```suggestion
		input := &ec2.AssignIpv6AddressesInput{
			NetworkInterfaceId: runResult.Instances[0].NetworkInterfaces[i].NetworkInterfaceId,
```